Zend/zend_execute.c

```diff
@@ -2022,6 +2022,10 @@ static void zend_pre_incdec_property_zval(zval *prop, zend_property_info *prop_i
 		} while (0);
 	}
 	if (UNEXPECTED(RETURN_VALUE_USED(opline))) {
+		/* Error handler can mess up properties and unset it */
+		if (UNEXPECTED(Z_TYPE_P(prop) == IS_UNDEF)) {
+			ZVAL_NULL(prop);
+		}
```

Comment:
I think this is wrong. We shouldn't change the property. In worst case this may be a typed property that can't be NULL. There are few similar places in the patch.

Reply:
@arnaud-lb gave me a solution that I am going to try out which should hopefully fix those in a sensible manner.